lightning/src/ln/features.rs

```diff
@@ -490,10 +490,11 @@ impl<T: sealed::Context> Features<T> {
 	/// Converts `Features<T>` to `Features<C>`. Only known `T` features relevant to context `C` are
 	/// included in the result.
 	fn to_context_internal<C: sealed::Context>(&self) -> Features<C> {
-		let byte_count = C::KNOWN_FEATURE_MASK.len();
+		let from_byte_count = T::KNOWN_FEATURE_MASK.len();
+		let to_byte_count = C::KNOWN_FEATURE_MASK.len();
 		let mut flags = Vec::new();
 		for (i, byte) in self.flags.iter().enumerate() {
-			if i < byte_count {
+			if i < from_byte_count && i < to_byte_count {
```

Comment:
Could you add a test case?

Reply:
I can later but won't have time this week. Asked Matt and he's fine with me PR'ing separately later, lmk what you think.

Reply:
No prob. I don't mind doing it if you want to get this merged in the meantime.

Reply:
That would be awesome! :pray: